include/private/tdydmimpl.h

```diff
@@ -0,0 +1,25 @@
+#if !defined(TDYDMIMPL_H)
+#define TDYDMIMPL_H
+
+#include <petsc.h>
+#include <private/tdyugdmimpl.h>
+#include <private/tdyugridimpl.h>
+
+typedef enum{
```

Comment:
How does this type relate to `TDyDiscretization`? It seems like the same struct with an additional `DM` field.